arangod/RocksDBEngine/RocksDBPersistedLog.h

```diff
@@ -49,17 +50,37 @@ struct RocksDBLogPersistor : std::enable_shared_from_this<RocksDBLogPersistor> {
   auto persist(std::shared_ptr<arangodb::replication2::replicated_log::PersistedLog> log,
                std::unique_ptr<arangodb::replication2::PersistedLogIterator> iter,
                WriteOptions const& options) -> futures::Future<Result>;
+  auto persist(std::shared_ptr<arangodb::replication2::replicated_log::PersistedLog> log,
+               replication2::LogIndex stop, WriteOptions const& options)
+      -> futures::Future<Result>;
 
   struct PersistRequest {
     PersistRequest(std::shared_ptr<arangodb::replication2::replicated_log::PersistedLog> log,
                    std::unique_ptr<arangodb::replication2::PersistedLogIterator> iter,
                    futures::Promise<Result> promise)
-        : log(std::move(log)), iter(std::move(iter)), promise(std::move(promise)) {}
+        : log(std::move(log)), action(InsertEntries{std::move(iter)}), promise(std::move(promise)) {}
+    PersistRequest(std::shared_ptr<arangodb::replication2::replicated_log::PersistedLog> log,
+                   replication2::LogIndex stop,
+                   futures::Promise<Result> promise)
+        : log(std::move(log)), action(RemoveFront{stop}), promise(std::move(promise)) {}
+
+    struct InsertEntries {
+      std::unique_ptr<arangodb::replication2::PersistedLogIterator> iter;
+    };
+
+    struct RemoveFront {
+      replication2::LogIndex stop;
+    };
```

Comment:
I would like to define an alias `using Action = std::variant<InsertEntries, RemoveFront>` and have a single `persist` method that takes such an action. (That way we we also need just a single `PersistRequest` constructor that takes that an action). IMHO that would also make the call sites of `persist` clearer.

Reply:
Great idea!